runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -241,40 +241,43 @@ local function on_line_impl(self, buf, line, is_spell_nav)
     end
 
     while line >= state.next_row do
-      local capture, node, metadata = state.iter()
+      local capture, node, metadata = state.iter(line)
 
-      if capture == nil then
-        break
+      local range = { root_end_row + 1, 0, root_end_row + 1, 0 }
+      if node then
```

Comment:
Do we need to adjust `next_row` if there is no node?

Reply:
I put it there just b/c it's in the spirit of the while loop to end with `next_row > line`.

It probably doesn't make a huge runtime difference though since even if you start over every line:

https://github.com/neovim/neovim/blob/master/runtime/lua/vim/treesitter/highlighter.lua#L238

you will get an empty list every time since you're starting from `line`.

So yeah, just aesthetics.